docs/proposals/20250513-propogate-taints.md

```diff
@@ -0,0 +1,563 @@
+or by adding them manually after Node creation.
+In case of CABPK updating taints currently also requires a rollout.
+
+This proposal aims to standardize the way users can define taints for nodes in workload Clusters as well as how they get propagated in case of changes.
+
+**Note:** CAPI already manages the `node.cluster.x-k8s.io/uninitialized` and `node.cluster.x-k8s.io/outdated-revision` taints today. This proposal does not plan to change this.
+
+### Goals
+
+- Make taints a first-class citizen in core Cluster API types.
+- Define how taints get propagated to corresponding `Node`'s.
+- Define how a migration from a bootstrap provider's implementation of taints to the new feature could look like (taking CABPK as example).
+- Taints managed by Cluster API should not interfere with taints applied by other actors.
+
+### Non-Goals/Future Work
+
+- Supporting taints on individual devices via [Dynamic Resource Allocation](https://kubernetes.io/docs/concepts/scheduling-eviction/dynamic-resource-allocation/#device-taints-and-tolerations). This may be added in the future, but is currently out of scope.
+- Change how CAPI manages the `node.cluster.x-k8s.io/uninitialized` and `node.cluster.x-k8s.io/outdated-revision` taints.
+
+## Proposal
+
+### User Stories
+
+#### Story 0
+
+In Kubernetes Clusters, taints and tolerations are what allow a workload to be scheduled to a specific Node.
+As a user, I would like to use this community-standard mechanism within the framework that Cluster API provides.
+
+#### Story 1
+
+As a user, I wish to use Cluster API to manage a set of Machines that have very specific characteristics for targeting workloads.
+Some examples of this might be:
+
+- Designating Nodes as `edge` Nodes and steering locality-critical workloads only to `edge` Nodes.
+- Designating Nodes as having a particular hardware capability, such as high performance GPUs
+
+#### Story 2
+
+As a user, I wish to have autoscaling capabilities using Kubernetes and Cluster API resources and conventions.
+I would like to define taints on a Cluster API resource representing some collection (including but not limited to MachineSets, MachinePools, and MachineDeployments).
+This is especially useful in scale-from-zero scenarios, where that autoscaling technology can reference taints on a collection to make decisions about the cloud resources available.
+
+#### Story 3
+
+As a user, I would like to update taint metadata on my collection resources without forcing a complete replacement of an owned resource, such as a Machine or Node.
+
+#### Story 4
+
+As a user, I want to define a taint that is applied only once so I could ensure workload is scheduled after a custom initialization process initialized the node and remove that taint.
+
+### Requirements (Optional)
+
+#### Functional Requirements
+
+Functional requirements are the properties that this design should include.
```

Comment:
Is this line part of the template rather than something we want in the merged doc?